### DISCUSSION TITLE
MySQL sharing for bungee networks

### DISCUSSION THREAD

**Discussion by dan28000** [Zrips/CMI#6030] [Enhancement]
Be able to share part of database for entire network
Like nicknames, mutes, kick, bans, usermeta, saved inventories, ip's etc... stuff that you would like to have on entire network
so like economy, homes, votifier, mail etc.. wont be shared - stuff that you want to be only on 1 server from network

MySQL right now is not that usefull only for speed and be able to easily edit stuff in DB

My idea is to have basically 2 tables for user data first per server and second to be shared and be able to configure what will go to shared one and per server

You would need to point every cmi server to same database and configure it to use diff main table prefix 

for example serverA: CMI_survival and serverB:CMI_skyblock would have main prefix and also have setting for shared prefix CMI_shared would be on both servers same
this could be implemented for all tables but mainly user data

this would fix like this bug if network would have shared data https://github.com/Zrips/CMI/issues/5966
also could be nice to fix [senderprefix] over bungee displaying placeholder it self
https://github.com/Zrips/CMI/issues/4812
https://github.com/Zrips/CMI/issues/4311
those two bug reports

**Comment by ScottGB** [comment]
I suggested this via Spigot DM to Zrips over a year ago (26 April 2020), where I was told it wouldn't happen because it would cause issues.

I still need it as much as I did back then, but I have no hope of it coming

**Reply by dan28000** [reply]
If this would happen and fixed a senderprefix over bungee in pm not working then CMI would be go to plugin for a entire bungee networks managing bans chat private msg and much more in this state its good enought for entire network but shared stuff for network would be beneficial a lot
Now there are features like ban system, pms if you want senderprefix in the format, nicknames, vanish and more that you can throw out of the window for bungee network

**Reply by Baikele** [reply]
I'm also looking for this plugin, did you find it?
